### PR TITLE
ADCM-6068: Implement base Accessor 

### PR DESCRIPTION
**Pull request by Starovoitov**

### CODE REVIEW

File: adcm_aio_client/core/accessors.py [hunk 1]
```diff
@@ -9,35 +9,170 @@
...
+    async def get_or_none(self: Self, predicate: T) -> T | None:
+        with suppress(ObjectDoesNotExistError):
+            obj = await self.get(predicate=predicate)
+            if obj:
+                yield obj
+            else:
+                yield None
+        yield None  # Default yield in case of exceptions
+
+    async def all(self: Self, page: int = 1, items: int = 10) -> List[T]:
```
Comment (Sealwing) [inline adcm_aio_client/core/accessors.py:149]: this function doesn't take arguments, it returns everything what's in there

File: adcm_aio_client/core/accessors.py [hunk 2]
```diff
@@ -9,35 +9,170 @@
...
+    async def iter(self: Self, offset: int = 1, limit: int = 10) -> AsyncGenerator[List[T]]:
+        paginator = self.paginate(offset, limit)
+        while True:
+            start, end = next(paginator)
+            response: RequesterResponse = await MockRequester.list(
+                self.path, self.query_param.update({"offset": start, "limit": end})
+            )
+            if start >= len(response.as_list()):
+                raise StopIteration
+            yield [self._create_object(object) for object in response.as_list()]
```
Comment (Sealwing) [inline adcm_aio_client/core/accessors.py:168]: same as for the all, also should yield exact objects, not lists of them

I believe all should just `return list(self.iter())`

File: adcm_aio_client/core/accessors.py [hunk 3]
```diff
@@ -9,35 +9,170 @@
...
 
+    async def filter(self: Self, **predicate) -> List[T]:
```
Comment (Sealwing) [inline adcm_aio_client/core/accessors.py:170]: please see if `list` method really introduces conflict in typing here, maybe we don't have to use List after all

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: adcm_aio_client/core/accessors.py [hunk 4]
```diff
@@ -9,35 +9,116 @@
...
     @abstractmethod
-    async def filter(self: Self, predicate: T) -> AsyncGenerator[List[T]]: ...
+    async def filter(self: Self) -> List[T]: ...
+
+    def _create_object(self: Self, data: Dict[str, Any]) -> T:
+        return self.class_type(requester=self.requester, data=data)  # type: ignore
+
+
+class PaginatedAccessor[T](Accessor):
+    def _gen_page_indexes(
```
Comment (Sealwing) [inline adcm_aio_client/core/accessors.py:53]: unused, i believe

**Review by a-alferov — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Pull request merged**